src/main/java/org/chameleoncloud/api/admin/UserGroupRolesResource.java

```diff
@@ -0,0 +1,384 @@
+package org.chameleoncloud.api.admin;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.SortedMap;
+import java.util.TreeMap;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import org.chameleoncloud.representations.UserGroupRoles;
+import org.jboss.resteasy.annotations.cache.NoCache;
+import org.keycloak.authorization.model.Policy;
+import org.keycloak.authorization.model.ResourceServer;
+import org.keycloak.authorization.model.Scope;
+import org.keycloak.authorization.store.PolicyStore;
+import org.keycloak.models.GroupModel;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.models.RealmModel;
+import org.keycloak.models.UserModel;
+import org.keycloak.representations.idm.authorization.GroupPolicyRepresentation;
+import org.keycloak.services.resources.admin.permissions.AdminPermissionEvaluator;
+import org.keycloak.services.resources.admin.permissions.AdminPermissionManagement;
+import org.keycloak.services.resources.admin.permissions.AdminPermissions;
+import org.keycloak.services.resources.admin.permissions.GroupPermissionManagement;
+
+public class UserGroupRolesResource {
+
+  private AdminPermissionEvaluator auth;
+  private KeycloakSession session;
+  private RealmModel realm;
+  private AdminPermissionManagement adminPermissionManagement;
+  private GroupPermissionManagement groupPermission;
+  private PolicyStore policyStore;
+
+  private final String GROUP_POLICY_TYPE = "group";
+
+  public UserGroupRolesResource(KeycloakSession session, AdminPermissionEvaluator auth,
+      RealmModel realm) {
+    this.auth = auth;
+    this.session = session;
+    this.realm = realm;
+    this.adminPermissionManagement = AdminPermissions.management(session, realm);
+    this.groupPermission = adminPermissionManagement.groups();
+    this.policyStore = adminPermissionManagement.authz().getStoreFactory().getPolicyStore();
+  }
+
+  private Set<String> getGroupIdsFromPolicyConfig(Map<String, String> policyConfig) {
+
+    // policy config returns a type of Map<String, String>
+    // e.g.
+    // {"groups":"[{\"id\":\"b770dfde-038b-493e-bc7f-a51a5ce86741\",\"extendChildren\":true}]"}
```

Comment:
I didn't catch this before, but if we know this is JSON, it is probably simpler to just parse it as JSON and then iterate through and grab the values of the "id" key rather than doing this pattern matching.